### PR TITLE
Copy Panktee + Translation + Transliteration

### PR DESCRIPTION
**Pull request by AkalUstat**
- [x] basic setup for shabads + display preferences
- [x] rework logic once again to support banis/ceremonies
- [x] support banis
- [x] fix some bugs with banis
- [x] attempt ceremonies?

## How it works
**_Please note that as of right now it currently gets whatever it can, bur prioritizes display preferences_**

### Steps
- Make sure that the shabad page (navigator page) is active Screen Shot 2019-07-07 at 11 54 28 AM
- Make sure the panktee that you want to copy is active
- `Ctrl/Cmd + c` to copy the following, if available: Panktee + translations + transliteration. 
### Example
This shabad: 
Screen Shot 2019-07-25 at 4 55 45 PM

With these display preferences: 
Screen Shot 2019-07-25 at 4 56 39 PM

Will copy: 
Screen Shot 2019-07-25 at 4 58 06 PM

### CODE REVIEW

**Comment by maneetpaul** [comment]
Continue working on this, but going to hold off on merging until after 5.2 is shipped

File: www/js/index.js [hunk 1]
```diff
@@ -172,6 +173,12 @@ if (typeof Mousetrap !== 'undefined') {
   Mousetrap.bind(['up', 'left'], prevLine);
   Mousetrap.bind(['down', 'right'], nextLine);
   Mousetrap.bind('space', spaceBar);
+  Mousetrap.bind('mod+c', () => {
+    if (document.activeElement.id === 'shabad-page') {
+      // copy.copyPanktee();
```
Comment (tsingh777) [inline www/js/index.js]: remove

File: www/js/toolbar.js [hunk 2]
```diff
@@ -130,6 +130,7 @@ const printCeremonies = rows => {
             global.core.search.loadCeremony(row.ID).catch(error => {
               analytics.trackEvent('ceremonyFailed', row.ID, error);
             });
+            global.core.copy.loadFromDB(row.ID, 'ceremony');
```
Comment (tsingh777) [inline www/js/toolbar.js]: `extrasTileFactory` needs this line too.

File: www/js/copy.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  let result;
+  if (idType === 'shabad') {
+    result = await baniDB.loadShabad(id);
+    remapShabad(result);
+  } else if (idType === 'bani') {
+    baniDB.loadBani(id, baniLengthMap[baniLength]).then(rows => {
+      remapBani(rows);
+    });
+  } else if (idType === 'ceremony') {
+    baniDB.loadCeremony(id).then(rows => {
```
Comment (tsingh777) [inline www/js/copy.js]: Ceremony isn't copying properly.

Comment (AkalUstat) [reply www/js/copy.js]: As in the explanation slides or shabads?

Comment (tsingh777) [reply www/js/copy.js]: shabads.

Comment (AkalUstat) [reply www/js/copy.js]: copying shabads in anand karaj working fine for me

Comment (tsingh777) [reply www/js/copy.js]: if you search for a shabad first, then load the ceremony and try to copy a line, it doesn't get the correct line.

Comment (AkalUstat) [reply www/js/copy.js]: seems like the error is coming from the fact that `search.currentShabad.indexOf(search.currentLine);` isnt updating correctly;

File: www/js/copy.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  });
+}
+/**
+ *
+ * @param {Objects} unmapped rows fresh from DB
+ * for each row, it sees if row.Verse (normal panktees) row.Custom (custom symbols or text) is available
+ * if not, it just displays title (because those two will only not be available at the position of the title)
+ * Then, it remapps the correct part of the row Object
+ */
+function remapBani(unmapped) {
```
Comment (AkalUstat) [inline www/js/copy.js]: @tsingh777 sukhmani sahib has an extra row.Verse at the beginning that offsets the rest of the bani by 1 (its only for sukhmani sahib). Whats the best way to fix that?

**Comment by maneetpaul** [comment]
This is very cool!

**Pull request merged**